pkg/kclient/pods.go

```diff
@@ -126,3 +129,21 @@ func (c *Client) ExecCMDInContainer(podName string, containerName string, cmd []
 
 	return nil
 }
+
+// ExtractProjectToComponent extracts the project archive(tar) to the target path from the reader stdin
+func (c *Client) ExtractProjectToComponent(compInfo common.ComponentInfo, targetPath string, stdin io.Reader) error {
+	// cmdArr will run inside container
+	cmdArr := []string{"tar", "xf", "-", "-C", targetPath}
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	glog.V(4).Infof("Executing command %s", strings.Join(cmdArr, " "))
+	err := c.ExecCMDInContainer(compInfo, cmdArr, &stdout, &stderr, stdin, false)
+	if err != nil {
+		glog.Errorf("Command '%s' in container failed.\n", strings.Join(cmdArr, " "))
+		glog.Errorf("stdout: %s\n", stdout.String())
+		glog.Errorf("stderr: %s\n", stderr.String())
+		glog.Errorf("err: %s\n", err.Error())
```

Comment:
![25810](https://user-images.githubusercontent.com/6422176/80248848-bc17eb80-863e-11ea-8e51-f4961f8b4394.jpg)

Reply:
errrr. maybe use fmt.Errorf? The `glog.Errorf` statements are out of place here.

Reply:
This has been there historically with odo occlient https://github.com/openshift/odo/pull/2626/files#diff-6025e9b620e9321c03c11bf29e5c4dbdL2717 but I'll update it

Reply:
using `log.Errorf` to print the stdout, stdErr, etc. since I think the original author of this code intended to do this, and returning all this info via `fmt.Errorf` will look dirty. Also looks like the exec code is historically doing this i.e.; they are `log.Errorf` the output on exec error..